nav2_navfn_planner/src/navfn_planner.cpp

```diff
@@ -148,12 +172,6 @@ nav_msgs::msg::Path NavfnPlanner::createPlan(
   if (start.pose.position.x == goal.pose.position.x &&
     start.pose.position.y == goal.pose.position.y)
   {
-    unsigned int mx, my;
-    costmap_->worldToMap(start.pose.position.x, start.pose.position.y, mx, my);
```

Comment:
Readd this - if in collision we should still fail, the trivial single-point paths should only be given if start is valid (because might need to rotate in place or something, and if already in collision, that's not good)!

Reply:
There's also an eq. check in Smac2D that is still there, both of these should now throw exceptions for path invalid due to collision (but not due to single valued path)

Reply:
For navfn, that issue should be caught above before it even gets to the equality check in line 150. 

I updated the check for smac2D.

Reply:
Good point!